lib/services/local/users.go

```diff
@@ -105,24 +105,30 @@ func (s *IdentityService) getUsersWithSecrets() ([]services.User, error) {
 	return users, nil
 }
 
-// CreateUser creates user if it does not exist
+// CreateUser creates user if it does not exist.
 func (s *IdentityService) CreateUser(user services.User) error {
 	if err := user.Check(); err != nil {
 		return trace.Wrap(err)
 	}
+
 	value, err := services.GetUserMarshaler().MarshalUser(user.WithoutSecrets().(services.User))
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
 	item := backend.Item{
 		Key:     backend.Key(webPrefix, usersPrefix, user.GetName(), paramsPrefix),
 		Value:   value,
 		Expires: user.Expiry(),
 	}
-	_, err = s.Create(context.TODO(), item)
-	if err != nil {
+
+	if _, err = s.Create(context.TODO(), item); err != nil {
+		if trace.IsAlreadyExists(err) {
```

Comment:
hm....my concern is that we are relying on the DB layer to properly handle "already exist" errors which may not be the case (given it could be implemented via "upsert" pattern).

Perhaps it would be safer to explicitly check if this user exists before trying to perform DB operation.

@awly what do you think?

Reply:
an explicit check is definitely more robust.
we should eventually settle on consistent DB operation semantics (either all upsert or all create/delete) to take the guesswork out

Reply:
revised to explicitly check users before update/create